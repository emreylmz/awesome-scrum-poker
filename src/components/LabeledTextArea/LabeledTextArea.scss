@import "../../styles/colors";
@import "../../styles/metrics";

.labeled-text-area {
  display: flex;
  flex-direction: column;
  flex: 0;
  align-content: center;

  &__label {
    margin: $default-margin;
  }
  textarea {
    margin: 0 $default-margin;
    height: 200px;
    font-size: 18px;
    padding: $default-margin;
    background-color: aliceblue;
    color: $primary-color;
    border-radius: $default-border-radius;
  }
}
