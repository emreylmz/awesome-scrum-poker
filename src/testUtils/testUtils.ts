export const emptyTestFunc = (): void => void 0
