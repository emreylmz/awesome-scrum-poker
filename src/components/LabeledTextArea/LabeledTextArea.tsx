import React from 'react'
import { Render } from '../../types'
import './LabeledTextArea.scss'

interface Props {
  label: string
  onChange?: (value?: string) => void
}

class LabeledTextArea extends React.PureComponent<Props> {
  private onChange = (event: any): void => {
    const value = event.target.value
    if (this.props.onChange) {
      this.props.onChange(value)
    }
  }
  render(): Render {
    const { label } = this.props
    return (
      <div className={'labeled-text-area'}>
        <div className={'labeled-text-area__label'}>{label}</div>
        <textarea onChange={this.onChange} />
      </div>
    )
  }
}

export type LabeledTextAreaProps = Props
export default LabeledTextArea
