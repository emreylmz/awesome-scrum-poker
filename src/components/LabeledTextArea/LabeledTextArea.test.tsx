import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import sinon from 'sinon'
import '../../testUtils/setupTests'
import LabeledTextArea, { LabeledTextAreaProps } from './LabeledTextArea'

type DoneCallback = jest.DoneCallback

const defaultProps: LabeledTextAreaProps = {
  label: 'My Text Area'
}

const getWrapper = (props?: LabeledTextAreaProps): ShallowWrapper => {
  return shallow(<LabeledTextArea {...defaultProps} {...props} />)
}
describe('Labeled Text Area UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should have a label', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    const labelWrapper = wrapper.find('.labeled-text-area__label')
    expect(labelWrapper).toBeDefined()
    expect(labelWrapper.text()).toEqual(defaultProps.label)
    done()
  })

  it('should change label title with props', (done: DoneCallback): void => {
    const testLabel = 'Test Label'
    const wrapper = getWrapper()
    let labelWrapper = wrapper.find('.labeled-text-area__label')
    expect(labelWrapper.text()).toEqual(defaultProps.label)
    wrapper.setProps({ label: testLabel })
    labelWrapper = wrapper.find('.labeled-text-area__label')
    expect(labelWrapper.text()).toEqual(testLabel)
    done()
  })
})

describe('Labeled Text Area Logic', (): void => {
  it('should call onChange method when its value changed', (done: DoneCallback): void => {
    const testFunc = sinon.spy()
    const testValue = 'Test Value'
    const wrapper = getWrapper({ ...defaultProps, onChange: testFunc })
    const labelWrapper = wrapper.find('textarea')

    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    labelWrapper.invoke('onChange')({ target: { value: testValue } })
    expect(testFunc.calledOnce).toBeTruthy()
    expect(testFunc.args[0][0]).toEqual(testValue)
    done()
  })
})
