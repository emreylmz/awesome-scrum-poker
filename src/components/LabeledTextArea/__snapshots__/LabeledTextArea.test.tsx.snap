// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeled Text Area UI should render correctly 1`] = `ShallowWrapper {}`;
